recipes/libcurl/all/test_package/conanfile.py

```diff
@@ -25,14 +21,11 @@ def build(self):
 
     @property
     def _test_executable(self):
-        if self.settings.os == "Windows":
-            return os.path.join("bin", "test_package.exe")
-        else:
-            return os.path.join("bin", "test_package")
+        return os.path.join(self.cpp.build.bindirs[0], "test_package")
```

Comment:
I simplified this after @SpaceIm remark here: https://github.com/conan-io/conan-center-index/pull/12377#pullrequestreview-1096719932

Unfortunately I now have a error [here](https://c3i.jfrog.io/c3i/misc/logs/pr/12956/1-linux-gcc/libcurl/7.77.0//66fe4987fe114c052c300157fb662fdbd5cea929-test.txt):

```
ERROR: libcurl/7.77.0 (test package): Error in test() method, line 22
	self.run(self._test_executable, run_environment=True)
while calling '_test_executable', line 18
	return os.path.join(self.cpp.build.bindirs[0], "test_package")
	TypeError: 'NoneType' object is not subscriptable
```

Reply:
Dunno if it's related but it's missing the test type and sf reference 

#12935 should be a decent example

Reply:
It doesn't help. I can't even reproduce it locally using the following docker image: `conanio/gcc5`